src/synapse/workload.py

```diff
@@ -359,11 +377,35 @@ def enable_federation_domain_whitelist(container: ops.Container, charm_state: Ch
     try:
         config = container.pull(SYNAPSE_CONFIG_PATH).read()
         current_yaml = yaml.safe_load(config)
-        if charm_state.synapse_config.federation_domain_whitelist is not None:
-            current_yaml["federation_domain_whitelist"] = [
-                item.strip()
-                for item in charm_state.synapse_config.federation_domain_whitelist.split(",")
-            ]
+        federation_domain_whitelist = charm_state.synapse_config.federation_domain_whitelist
+        if federation_domain_whitelist is not None:
+            current_yaml["federation_domain_whitelist"] = _create_tuple_from_string_list(
+                federation_domain_whitelist
+            )
+            container.push(SYNAPSE_CONFIG_PATH, yaml.safe_dump(current_yaml))
```

Comment:
Since we do this pull, update, push operation over and over, perhaps we can have a helper method that does this? That way we reduce the repeated code

Reply:
I'll mark this one to be done later as a refactoring, thanks.

Reply:
ISD-1422